### ISSUE TITLE
Symlink Persists

### ISSUE THREAD

**Issue by sachnk** [apptainer/singularity#563]
Hi --

We're using Singularity in Penguin's POD cluster with Cent OS 7. We ran into a bizarre problem as described below.

For legacy reasons, we require the symlink  /home/myuser -> /mnt to exist in our Singularity container.

The following works as expected:

`singularity shell -B /path/to/data:/mnt myimage.img`
`$ ln -s /mnt /home/myuser`

In other words, when I `ls /home/myuser`, I see the contents of /path/to/data indirectly via /mnt.

This works as expected; every time I bring up the Singularity image, I always create the symlink -- it will *not* persist across Singularity restarts.

To run our jobs on the cluster, we wrap our job in a bootstrap script that first always creates the symlink as described above. Bizarrely enough, we notice that after we schedule these jobs to run and let them all finish, the symlink /home/myuser is now always available whenever we shell into our image.

In other words, the symlink now persists across singularity restarts.

It's actually okay for us if this is expected behavior, but I suspect it's not. Just want to understand what might be going on here?

Thanks!

**Comment by vsoch** [comment]
Cool! I mean, maybe not, haha. Just curious - the symlink is created between two directories (/mnt and a home folder) that are mounted to the image, correct? I think symbolic links are sort of like metadata stored with these [inode](https://en.wikipedia.org/wiki/Inode) things. So if you are creating a link between two files that are mounted on the host, it wouldn't be such a crazy idea that the inode of the host is storing the link, and the image mount is simply reading that. I bet you it's possible to keep this inode entry even when the mount goes away, although I'd need to read more about it. Did you check if the symlink persists on the host?

**Comment by jscook2345** [comment]
@sachnk Some containers are writeable. It is possible that something modified it during a step. It's hard to say without seeing the definition file and your build instructions. Can you share those?

Additionally, If this is no longer an issue for you, please let me know! Thanks

**Comment by dtrudg** [comment]
Hello - this issue hasn't received a reply for 12 months so I'm going to close it as stale.

Please feel free to reopen if the issue exists in the latest version of Singularity. Thanks!
